### PR TITLE
feat: support vacuum temporary table

### PR DESCRIPTION
**Pull request by SkyFan2002**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary
1. Support vacuum files of temporary table(Only tested in mysql session).
2. Fix copied files deduplication error of tmeporary table.
3. Change some logs from `info` to `debug`. 
4. Add `system.temporary_tables`. 

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16364)

### CODE REVIEW

**Comment by BohuTANG** [comment]
@dosu can you help me review this PR?

**Comment by BohuTANG** [comment]
@dosubot can you help me review this PR?

File: tests/sqllogictests/suites/temp_table/temporary_tables_table.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+statement ok
+CREATE TEMP TABLE d2.t2(a int not null, b int not null) ENGINE = Memory;
+
+query T
+select * from system.temporary_tables order by table_id;
+----
+default t0 4611686018427407904 FUSE
+d1 t1 4611686018427407905 FUSE
+d2 t2 4611686018427407906 MEMORY
```
Comment (dantengsky) [inline tests/sqllogictests/suites/temp_table/temporary_tables_table.sql:21]: maybe we should exclude the `table_id` column? likely to be flaky?

Comment (SkyFan2002) [reply tests/sqllogictests/suites/temp_table/temporary_tables_table.sql:21]: The table_id for temporary tables is allocated by incrementing from a fixed starting value, which should be deterministic in this case.

Comment (dantengsky) [reply tests/sqllogictests/suites/temp_table/temporary_tables_table.sql:21]: got it

**Review by dantengsky — APPROVED** [review]
look good to me @youngsofun PTAL, thanks

File: src/query/storages/common/session/src/temp_table.rs [hunk 2]
```diff
@@ -198,7 +198,7 @@ impl TempTblMgr {
         let desc = format!("{}.{}", database_name, table_name);
         let id = self.name_to_id.get(&desc);
         let Some(id) = id else {
```
Comment (BohuTANG) [inline src/query/storages/common/session/src/temp_table.rs:200]: This debug code no need anymore?

File: src/query/storages/fuse/src/table_functions/fuse_vacuum_temporary_table.rs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            let path = entry.path();
+            debug!("path: {}", path);
+            if let Some(session_id) = path.split('/').nth(1) {
+                if session_id.is_empty() {
+                    continue;
+                }
+                // check if session_id is a valid uuid
+                let _ = Uuid::parse_str(session_id)
+                    .map_err(|e| ErrorCode::Internal(format!("Invalid session_id: {}", e)))?;
+                debug!("session_id: {}", session_id);
```
[the remark this answers is not in the document]
Comment (SkyFan2002) [reply src/query/storages/fuse/src/table_functions/fuse_vacuum_temporary_table.rs]: got it

File: tests/sqllogictests/suites/temp_table/temporary_tables_table.sql [hunk 4]
```diff
@@ -0,0 +1,22 @@
+statement ok
+CREATE TEMP TABLE t0(a int not null, b int not null);
```
Comment (TCeason) [inline tests/sqllogictests/suites/temp_table/temporary_tables_table.sql:2]: Why this test do not add `onlyif mysql` tag?

Comment (SkyFan2002) [reply tests/sqllogictests/suites/temp_table/temporary_tables_table.sql:2]: https://github.com/datafuselabs/databend/blob/f885364b8c4eefe804534e007637b9378fe4ea36/scripts/ci/ci-run-sqllogic-tests.sh#L22-L25

Comment (TCeason) [reply tests/sqllogictests/suites/temp_table/temporary_tables_table.sql:2]: https://github.com/datafuselabs/databend/blob/f885364b8c4eefe804534e007637b9378fe4ea36/scripts/ci/ci-run-sqllogic-tests.sh#L23C1-L24C1

File: src/query/service/src/sessions/query_ctx.rs [hunk 5]
```diff
@@ -1335,8 +1335,17 @@ impl TableContext for QueryContext {
...
+    fn get_session_id(&self) -> Result<String> {
+        let session_type = self.shared.session.get_type();
+        match session_type {
+            SessionType::MySQL => Ok(self.shared.session.id.clone()),
+            _ => self
+                .shared
+                .session
+                .session_ctx
+                .get_client_session_id()
+                .ok_or(ErrorCode::Internal("No client session id".to_string())),
```
Comment (TCeason) [inline src/query/service/src/sessions/query_ctx.rs:1347]: Why disable http SessionType?

**Pull request merged**